clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2388,11 +2388,40 @@ static void handleCHERIMethodSuffix(Sema &S, Decl *D, const ParsedAttr &Attr) {
   D->addAttr(::new (S.Context) CHERIMethodSuffixAttr(S.Context, Attr, Str));
 }
 
-static void handleCHERICompartmentName(Sema &S, Decl *D, const ParsedAttr &Attr) {
+static void handleCHERICompartmentName(Sema &S, Decl *D, const ParsedAttr &Attr,
+                                       Sema::DeclAttributeLocation DAL) {
+  // cheri_compartment is both:
```

Comment:
I think we need to handle the attribute only at the DeclSpec level and not in the DeclChunk one. Otherwise, it generates duplicate messages.

Reply:
Oh that's the part that generated the linking error ! I'll try to recreate the function declaration/definition where the attribute is skipped by this early return.

Reply:
I think I've got the gotcha. An attribute after a pointer return type remains tied to the DeclChunk (`void * __attribute__((...))`). So my code was wrong.

Reply:
maybe I should do the way round, ie only process the DeclChunk.